src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,31 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final byte MIN_LOG_PASSWORD_LENGTH = 6;
+    private static final byte MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("User with the same login already exists");
+        }
+        if (user.getLogin() == null
+                || user.getLogin().length() < MIN_LOG_PASSWORD_LENGTH) {
+            throw new InvalidDataException("Login must be at least 6 characters");
+        }
+        if (user.getPassword() == null
```

Comment:
Check for a null password should be a separate if statement to provide a clear and specific error message.